src/src_/scan.toit

```diff
@@ -1,51 +1,71 @@
+// Copyright (C) 2024 Toitware ApS.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the package's LICENSE file.
+
+import ble
 import net.wifi
 import protobuf
 
-import .ble show Process_
-import .utils
+import .rpc
+import .security
 import ..proto_.constants-pb
 import ..proto_.wifi-scan-pb
 
-class ScanProcess_ implements Process_:
+/**
+An RPC service responsible for scanning for WiFi access points.
+*/
+class ScanRpcService extends ProtobufRpcService:
+  static ID_ ::= 0x50
+  static DESCRIPTION_ ::= "prov-scan"
+
   static CHANNEL-NUM ::= 14
   static SCAN-AP-MAX ::= 16
 
-  ap-list/List := []
+  /**
+  A list of access points that have been found so far.
+  */
+  ap-list_/List := []
+  scan-task_/Task? := null
+
+  constructor service/ble.LocalService --security/Security:
+    super service ID_ --description=DESCRIPTION_ --security=security
+    ap-list/List := []
+
+  close -> none:
+    critical-do:
+      stop-scanning_
+      super
 
-  compare-ap-by-rssi a/wifi.AccessPoint b/wifi.AccessPoint -> int:
+  static compare-ap-by-rssi_ a/wifi.AccessPoint b/wifi.AccessPoint -> int:
     return -(a.rssi.compare-to b.rssi)
```

Comment:
Isn't this a bit weird? Wouldn't it be more common to do `b.rssi.compare-to a.rssi`?

Reply:
I think I actually prefer this approach. Makes it clear that it is inverted.